support/k8s/k8s-workload-registrar/mode-crd/controllers/identity_schema.go

```diff
@@ -0,0 +1,313 @@
+	podUIDLabel         = "PodUID"
+	podNameLabel        = "PodName"
+	serviceAccountLabel = "ServiceAccount"
+)
+
+type IdentitySchema struct {
+	Version string  `yaml:"version"`
+	Fields  []Field `yaml:"fields"`
+}
+
+type Field struct {
+	Name            string           `yaml:"name"`
+	AttestorSource  *AttestorSource  `yaml:"attestorSource,omitempty"`
+	ConfigMapSource *ConfigMapSource `yaml:"configMapSource,omitempty"`
+}
+
+type Source interface {
+	GetValue(pod *corev1.Pod) (string, error)
+}
+
+type ConfigMapSource struct {
+	Name      string `yaml:"name"`
+	Namespace string `yaml:"ns"`
+	Field     string `yaml:"field"`
+}
+
+type AttestorSource struct {
+	Name    string    `yaml:"name"`
+	Group   string    `yaml:"group"`
+	Mapping []Mapping `yaml:"mapping"`
+}
+
+type Mapping struct {
+	Type  string `yaml:"type"`
+	Field string `yaml:"field"`
+}
+
+func (is *IdentitySchema) loadConfig(fileName string) (*IdentitySchema, error) {
+
+	log.Print("identity_schema, loadConfig: before read")
+	yamlFile, err := ioutil.ReadFile(fileName)
+	if err != nil {
+		log.Printf("identity_schema, loadConfig: Error reading yaml file %s:  %v ", fileName, err)
+		return is, err
+	}
+	log.Print("identity_schema, loadConfig: after read")
+
+	err = yaml.Unmarshal(yamlFile, is)
+	if err != nil {
+		//log.Fatalf("Unmarshal: %v", err)
+		log.Printf("identity_schema, loadConfig: Error processing YAML file %v", err)
+		return is, err
+	}
+	return is, nil
+}
+
+func main() {
+	var is IdentitySchema
+
+	// if err := r.Get(ctx, req.NamespacedName, &pod); err != nil {
+
+	if _, err := is.loadConfig("/tmp/identity-schema.yaml"); err != nil {
+		log.Fatalf("Error getting IdenitySchema config %v", err)
+	}
+
+	// Set up pod:
+	pod := &corev1.Pod{
+		Spec: corev1.PodSpec{
+			ServiceAccountName: "podServiceAccount",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace:   "podNamespace",
+			Labels:      map[string]string{},
+			Annotations: map[string]string{},
+		},
+	}
+	// if testCase.configLabel != "" && testCase.podLabel != "" {
+	// 	pod.Labels[testCase.configLabel] = testCase.podLabel
+	// }
+	// if testCase.configAnnotation != "" && testCase.podAnnotation != "" {
+	// 	pod.Annotations[testCase.configAnnotation] = testCase.podAnnotation
+	// }
+
+	// Test:
+	//spiffeID := c.podSpiffeID(pod)
+
+	finalId := is.getSVID(pod)
+	log.Printf("** Final id %v", finalId)
+	log.Printf("Identity %#v", is)
+	fmt.Print(&is)
+}
+
+func (is *IdentitySchema) getSVID(pod *corev1.Pod) string {
+
+	log.Printf("identity_schema, getId: processing Pod %s", pod.Name)
+
+	var idString string = ""
+	fields := is.Fields
+	for i, field := range fields {
+		log.Printf("identity_schema, getId: %d Field name: %s", i, field.Name)
+		idString += "/" + is.getFieldValue(pod, field)
+	}
+	log.Printf("identity_schema, getSVID: ID Value: %s", idString)
+	return idString
+}
+
+func (is *IdentitySchema) getFieldValue(pod *corev1.Pod, field Field) string {
+
+	att := field.AttestorSource
```

Comment:
may be cleaner to use a switch statement, and also will check if we miss  any cases like https://github.com/kubernetes/kubernetes/blob/v1.21.1/pkg/volume/plugins.go#L495-L502

Reply:
Oh, that's a very nice and clean method. Thank you for the suggestion!